2주차/11866/java_11866_루키.java

```diff
@@ -0,0 +1,52 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringTokenizer;
+
+public class java_11866_루키 {
+
+    public static void main(String[] args) throws IOException {
+
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringTokenizer st = new StringTokenizer(br.readLine());
+
+        int N, K;
+        N = Integer.parseInt(st.nextToken());
+        K = Integer.parseInt(st.nextToken());
+
+        List<Integer> numbers = new ArrayList();
+        for (int i = 1; i <= N; i++) {
+            numbers.add(i);
+        }
+
+        int count = 1;
+        int index = 0;
+
+        List<Integer> answers = new ArrayList<>();
+        while (numbers.size() != 0) {
+            if (numbers.size() == index) {
+                index = 0;
+                continue;
+            }
+            if (count == K) {
+                answers.add(numbers.remove(index));
+                index--;
+                count = 0;
+            }
+            if (count != K) {
+                index++;
+                count++;
+            }
+        }
+
+        System.out.print("<");
+        for (int i = 0; i < answers.size() - 1; i++) {
+            System.out.print(answers.get(i));
+            System.out.print(", ");
+        }
+        System.out.print(answers.get(answers.size() - 1));
+        System.out.print(">");
```

Comment:
IO작업의 경우 리소스를 많이 잡아 먹기때문에 입력에서 `BufferedReader`를 사용한 이유와 같이 출력 쪽도 신경을 써주시면 속도 측면에서 이점을 볼 수 있어요!

결과로 여러 출력을 하는 경우에는 아래의 방법들을 사용할 수 있을거 같아요.
1. StringBuilder를 통해 하나의 정답 String을 가공하고 `System.out.println()`을 사용하는 방법
2. BufferedWriter를 사용해 버퍼에 결과를 저장 후, 한번만 flush하는 방법

아래의 자료를 참고해보시면 좋을것 같습니다!!
- [[Java] 빠른 입출력을 위한 BufferedReader, BufferedWriter, StringTokenizer, StringBuilder](https://rlakuku-program.tistory.com/33)

Reply:
```suggestion
        StringBuilder sb = new StringBuilder();
        sb.append("<");
        for (int i = 0; i < answers.size() - 1; i++) {
            sb.append(answers.get(i));
            sb.append(", ");
        }
        sb.append(answers.get(answers.size() - 1));
        sb.append(">");

        System.out.println(sb);
```
위와 같이 StringBuilder를 통해 sout 한번만 하게 될 경우 아래와 같이 성능 향상을 얻을 수 있어요!

<img width="936" alt="image" src="https://user-images.githubusercontent.com/57744251/178137798-44434796-8ffc-433f-8d1e-39766848c906.png">
아래의 결과는 기존 루키의 코드 실행 결과이고, 위의 결과는 위의 suggestion대로 수정한 후에 실행한 결과입니다.